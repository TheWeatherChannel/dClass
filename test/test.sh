#!/bin/bash

OPENDDR_RESOURCE=/some/path/OpenDDR/latest/resources
DCLASS_BIN=../src/dclass_client
#DCLASS_BIN="../java/dclass.jar"

if [ ! -s "uas.txt" ]
then
        echo "ERROR: uas.txt not found"
        exit 1
fi

if [ ! -s "$DCLASS_BIN" ]
then
        echo "ERROR: dclass_client binary not found"
        exit 1
fi

#DCLASS_BIN="java -cp ../java/dclass.jar dclass.dClassMain"

CFLAG=`$DCLASS_BIN test.txt | grep "UA lookup 5: '" | wc -l`

if [ $CFLAG != "1" ]
then
	echo "ERROR: binary not compiled correctly"
	echo "Make sure DTREE_TEST_UALOOKUP is set to 1"
	exit 1
fi

if [ -d "$OPENDDR_RESOURCE" ]
then
	$DCLASS_BIN -d "$OPENDDR_RESOURCE" uas.txt | grep "^UA lookup " | sed "s/time.*$//" > uas.openddr.testd

	TESTDDIFF=`diff uas.openddr uas.openddr.testd | grep "^<" | sed "s/^.* lookup //" | sed "s/: '.*$//" | xargs echo`

	if [ "$TESTDDIFF" != "" ]
	then
		echo "ERROR: Test OpenDDR failed"
		echo "Lines in uas.openddr: $TESTDDIFF"
		echo "Move uas.openddr.testd to uas.openddr if test is correct"
		exit 1
	else
		echo "PASSED: Test OpenDDR"
	fi
else
	echo "SKIPPED: Test OpenDDR"
fi

$DCLASS_BIN -l ../dtrees/openddr.dtree -o test2.dtree uas.txt | grep "^UA lookup " | sed "s/time.*$//" > uas.openddr.test1

TEST1DIFF=`diff uas.openddr uas.openddr.test1 | grep "^<" | sed "s/^.* lookup //" | sed "s/: '.*$//" | xargs echo`

if [ "$TEST1DIFF" != "" ]
then
	echo "ERROR: Test 1 failed"
	echo "Lines in uas.openddr: $TEST1DIFF"
	echo "Move uas.openddr.test1 to uas.openddr if test is correct"
	exit 1
else
	echo "PASSED: Test 1"
fi

$DCLASS_BIN -l test2.dtree uas.txt | grep "^UA lookup " | sed "s/time.*$//" > uas.openddr.test2

TEST2DIFF=`diff uas.openddr uas.openddr.test2 | grep "^<" | sed "s/^.* lookup //" | sed "s/: '.*$//" | xargs echo`

if [ "$TEST2DIFF" != "" ]
then
	echo "ERROR: Test 2 failed"
	echo "Lines: $TEST2DIFF"
	exit 1
else
	echo "PASSED: Test 2"
fi

$DCLASS_BIN -l ../dtrees/browser.dtree uas.txt | grep "^UA lookup " | sed "s/time.*$//" > uas.browser.test3

TEST3DIFF=`diff uas.browser uas.browser.test3 | grep "^<" | sed "s/^.* lookup //" | sed "s/: '.*$//" | xargs echo`

if [ "$TEST3DIFF" != "" ]
then
	echo "ERROR: Test 3 failed"
	echo "Lines: $TEST3DIFF"
	echo "Move uas.browser.test3 to uas.browser if test is correct"
	exit 1
else
	echo "PASSED: Test 3"
fi

$DCLASS_BIN -l ../dtrees/test.dtree test.txt -o test5.dtree | grep "^UA lookup " | sed "s/time.*$//" > test.out.test4

TEST4DIFF=`diff test.out test.out.test4 | grep "^<" | sed "s/^.* lookup //" | sed "s/: '.*$//" | xargs echo`

if [ "$TEST4DIFF" != "" ]
then
	echo "ERROR: Test 4 failed"
	echo "Lines: $TEST4DIFF"
	echo "Move test.out.test4 to test.out if test is correct"
	exit 1
else
	echo "PASSED: Test 4"
fi

$DCLASS_BIN -l test5.dtree test.txt | grep "^UA lookup " | sed "s/time.*$//" > test.out.test5

TEST5DIFF=`diff test.out test.out.test5 | grep "^<" | sed "s/^.* lookup //" | sed "s/: '.*$//" | xargs echo`

if [ "$TEST5DIFF" != "" ]
then
	echo "ERROR: Test 5 failed"
	echo "Lines: $TEST5DIFF"
	exit 1
else
	echo "PASSED: Test 5"
fi

rm uas.openddr.test* uas.browser.test* test2.dtree test.out.test* test5.dtree > /dev/null 2>&1

echo "All tests completed"

exit 0
