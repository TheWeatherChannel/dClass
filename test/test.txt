abc123
xy1234
xyc123
aaxyzbbb
3 2 1 three two one 3 2 1
0ne test two0 test 3 test
1 2 one_tw0 three four
rat vat bat sat mat cat nat
rat vat bat sat mat nat
rat vat sat mat nat
rat vat sat nat
rat vat nat
rat nat
nat
test3 test4
test4 test3
test2 test1
test1 test2
test5 test6
test6 test5
test7 test9 test8
test7 test8 test9
settest
set.test
set1test
set..test
set\test
set\\test
group0 group1 group2 group3 group4
group0 group1) group t3st? group3 group4
group0 group1) group tstx group3 group4
group0 group1) group tstxx group3 group4
pattern 123  456
pattern 123  456 
pattern 123  "  456 
a ptest b c d e f
ptest b c d e f
a b ptest c d e f
a b c ptest d e f
a b c d e f g ptest
a b c d e f g h i j k ptest
a b fullstring c
a fullstring
fullstring
a full string
full string
full 55 b string
full string b
some wtest25 text wtest5b wtest wtest100
some wtest25 text wtest5b wtest wtest10x0
some wtest5 text wtest5b wtest wtest10x0
some wtest5 text wtestb wtest wtest10x0
some wtestx text wtestb wtest wtest10x0
3 this is the end x
4 this is the start x
5 this is the start end
5 this is the middle end
aaa bbb ccc
aaa bb ccc bbb
aaa bbb ccc bbb
aaa bbb 5 4 3 2 ccc
aaa bbb 5 4 3 2 ccc bbb
aaa bbb 5 4 3 2 ccc 7 6 bbb
aaa bb 5 4 3 2 ccc 7 6 bbb
del,del=del
quote";"quote
