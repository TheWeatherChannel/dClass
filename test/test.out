UA lookup 1: 'abc' 
UA lookup 2: 'xyz' 
UA lookup 3: 'xyz' 
UA lookup 4: 'xyz' 
UA lookup 5: '123' 
UA lookup 6: '123' 
UA lookup 7: '123s' 
UA lookup 8: 'cat' 
UA lookup 9: 'bat' 
UA lookup 10: 'mat' 
UA lookup 11: 'sat' 
UA lookup 12: 'vat' 
UA lookup 13: 'rat' 
UA lookup 14: 'unknown' 
UA lookup 15: 'test34' 
UA lookup 16: 'test3' 
UA lookup 17: 'test1' 
UA lookup 18: 'test12' 
UA lookup 19: 'test56' 
UA lookup 20: 'test5' 
UA lookup 21: 'test8' 
UA lookup 22: 'test89' 
UA lookup 23: 'settest2' 
UA lookup 24: 'settest' 
UA lookup 25: 'settest2' 
UA lookup 26: 'settest' 
UA lookup 27: 'settest3' 
UA lookup 28: 'unknown' 
UA lookup 29: 'g123' 
UA lookup 30: 'g123' 
UA lookup 31: 'g123' 
UA lookup 32: 'unknown' 
UA lookup 33: 'unknown' 
UA lookup 34: 'p123456n' 
UA lookup 35: 'p123456s' 
UA lookup 36: 'ptest2' 
UA lookup 37: 'ptest1' 
UA lookup 38: 'ptest3' 
UA lookup 39: 'unknown' 
UA lookup 40: 'pteste' 
UA lookup 41: 'pteste' 
UA lookup 42: 'fullsa' 
UA lookup 43: 'fullsa' 
UA lookup 44: 'fulls' 
UA lookup 45: 'unknown' 
UA lookup 46: 'sfulls' 
UA lookup 47: 'sfulls2' 
UA lookup 48: 'unknown' 
UA lookup 49: 'wtest100' 
UA lookup 50: 'wtest25' 
UA lookup 51: 'wtest5b' 
UA lookup 52: 'wtest5' 
UA lookup 53: 'wtest' 
UA lookup 54: 'unknown' 
UA lookup 55: 'thiss' 
UA lookup 56: 'thise' 
UA lookup 57: 'middle' 
UA lookup 58: 'bbbb' 
UA lookup 59: 'bbba' 
UA lookup 60: 'bbba' 
UA lookup 61: 'bbbbm' 
UA lookup 62: 'bbba' 
UA lookup 63: 'bbbbm' 
UA lookup 64: 'bbbam' 
UA lookup 65: 'delt' 
UA lookup 66: 'quote' 
